app/api/csv/specs/typeParsers.spec.js

```diff
@@ -110,4 +110,120 @@ describe('csvLoader typeParsers', () => {
       }
     );
   });
+
+  describe('multidate', () => {
+    it.each`
+      dateProp                                 | dateFormat      | expectedDate
```

Comment:
Maybe also test: 
'' (empty string)
'01-01-2014'
'01-01-2014||01-01-2015' (double separator in the middle)
'2014|2015|' (terminate with separator)

Reply:
Added some extra cases.